sdk/cognitiveservices/azure-cognitiveservices-language-textanalytics/azure/cognitiveservices/language/textanalytics/_response_handlers.py

```diff
@@ -0,0 +1,230 @@
+    """
+    raise_error = HttpResponseError
+    error_message = error.error["innerError"]["message"]
+    error_code = error.error["code"]
+    error_message += "\nErrorCode:{}".format(error_code)
+    error = raise_error(message=error_message)
+    error.error_code = error_code
+    raise error
+
+
+def process_batch_error(error):
+    """Raise and return detailed error message for HttpResponseErrors
+    """
+    raise_error = HttpResponseError
+    if error.status_code == 401:
+        raise_error = ClientAuthenticationError
+    error_message = error.message
+    error_code = error.status_code
+    error_body = None
+
+    try:
+        error_body = ContentDecodePolicy.deserialize_from_http_generics(error.response)
+    except DecodeError:
+        pass
+
+    if error_body is not None:
+        error_resp = error_body["error"]
+        try:
+            error_message = error_resp["innerError"]["message"]
+            error_code = error_resp["innerError"]["code"]
+        except KeyError:
+            error_message = error_resp["message"]
+            error_code = error_resp["code"]
+
+        error_message += "\nErrorCode:{}".format(error_code)
+
+    error = raise_error(message=error_message, response=error.response)
+    error.error_code = error_code
+    raise error
+
+
+def _validate_single_input(text, hint, hint_value):
+    """Validate text input is string. Let service handle
+    validity of hint and hint value.
+
+    :param str text: A single text document.
+    :param str hint: Could be country_hint or language
+    :param str hint_value: The user passed country_hint or language
+    :return: A LanguageInput or MultiLanguageInput
+    """
+    if isinstance(text, six.string_types):
+        if hint_value:
+            return [{"id": "0", "text": text, hint: hint_value}]
+        return [{"id": "0", "text": text}]
+    raise TypeError("Text parameter must be string.")
+
+
+def _validate_batch_input(documents, hint, hint_value):
+    """Validate that batch input has either all string docs
+    or dict/LanguageInput/MultiLanguageInput, not a mix of both.
+
+    :param list documents: The input documents.
+    :return: A list of LanguageInput or MultiLanguageInput
+    """
+    if not isinstance(documents, list):
+        raise TypeError("Documents parameter must be a list.")
+
+    string_input, nonstring_input = False, False
+    string_batch = []
+    for idx, doc in enumerate(documents):
+        if isinstance(doc, six.string_types):
+            if hint_value:
+                string_batch.append({"id": str(idx), hint: hint_value, "text": doc})
+            else:
+                string_batch.append({"id": str(idx), "text": doc})
+            string_input = True
+        if isinstance(doc, (dict, MultiLanguageInput, LanguageInput)):
+            nonstring_input = True
+
+    if string_input and nonstring_input:
+        raise TypeError("Mixing string and dictionary/object input unsupported.")
```

Comment:
Instead of interleaving the mixing detection throughout this code, might suggest moving the check to the top:
```
if not (all(isinstance(x, six.string_types) for x in documents):
    if not all(isinstance(x, (dict, MultiLanguageInput, LanguageInput)) for x in documents):
        raise(...)
```
If there is a performance consideration (i.e. if `documents` is usually very long) that would be an affirmative reason to do things in one pass